### DISCUSSION TITLE
thoughts about ghostty? 👻

### DISCUSSION THREAD

**Discussion by lovelindhoni** [wezterm/wezterm#6520] [General]
I built and tried Ghostty today, and I could see no reason to switch from WezTerm to it. I just find it all overhyped. What do you guys think?

**Comment by yriveiro** [comment]
I tried it yesterday, and I feel nothing fantastic about it more than filling it like a native application.

- I couldn't configure the tab style and information (or at least I don't know how to)
- I need to return to using Tmux. While I love it, I was only using its Pane and Session management features, which Wezterm already provides
- Its image rendering isn't better than Wezterm's, and when using Neovim + image.vim, Ghostty crashes.
- I'm biased regarding configuration since I'm accustomed to Lua scripting

Regarding performance, while I don't find Wezterm slow, there's always room for improvement. Areas like the image protocol and plugin system could be improved as well

One advantage Ghostty might have is its potentially larger contributor base compared to Wezterm. Wezterm project has 104 PR open and 1.1K issues.

I don't see the ROI of moving to Ghostty, the only reason to do it is if Wezterm stops in time or gets too heavy as iTerm2 did.

**Comment by habahabahaba** [comment]
1. On macOS, Ghostty has secure keyboard entry.
2. According to these (quite extensive) tests, Ghostty is 2 to 5 times more performant than Wezterm, depending on the scenario.

**Reply by yriveiro** [reply]
@habahabahaba 

- Secure keyboard entry is a nice feature indeed.
- Regarding perf. Performance is bonded to user context in the sense that if the user doesn't feel the lag, that performance has no intrinsic value, others for sure will switch, I went from iTerm2 to Alacritty even Alacritty lacked almost everything but perf. At the time programming with Neovim on iTerm2 was unbearable with large files.

**Reply by quantonganh** [reply]
Since I'm using WezTerm as part of my setup to transform Helix into an IDE, there are certain features that I can't live without: `get-text`, `set-text`, `split-pane`, `get-pane-direction`, quick select, [floating pane](https://github.com/wez/wezterm/pull/5576), among others.

Essentially, I rely on an [API for managing splits](https://github.com/ghostty-org/ghostty/discussions/2353) to optimize my workflow.

> According to these (quite extensive) tests, Ghostty is 2 to 5 times more performant than Wezterm, depending on the scenario.

While Ghostty does perform well in real-world usage, I found the benchmarking results surprising when testing with [DOOM-fire-zig](https://github.com/const-void/DOOM-fire-zig):
- wezterm: ~400 fps
- ghostty 1.0: ~160 fps
- ghostty 1.0.1: ~250 fps

I'm not sure what might be causing this.
Could you try running the benchmark and share your results? Thanks!

**Reply by habahabahaba** [reply]
On my M2 Mac-Air Sonoma 14.7:
Ghostty 1.01 - 515 fps,
Wezterm 20240203-110809-5046fc22 - 430 fps.
Not that big of a gap in this case.

**Comment by mckcd** [comment]
while I like ghostty's config simplicity  i find the only thing it has over wezterm or other terminals is the terminal inspector 
which is not that useful now but i think in future it will be for developing terminal apps etc

**Comment by bkw777** [comment]
It's garbage. Haphazard coding that works for some people by luck. I especially love the curl sudo bash install directions for ubuntu.
My own particular example: https://github.com/ghostty-org/ghostty/discussions/5549
"Works for me. Anyway we also changed it so it's prolly fixed now...closed" Meanwhile not only did the original problem not get fixed at all, it's even worse now with new bugs added on top. Complete junk.

I have no use for that project, I was only testing it because I have an app that uses a terminal window. It has a built-in libvte terminal but I also allow the user to configure whatever terminal they want, so I try to test all possible terminals to figure out what command line args they need and how to deal with their different quirks. So far this is the only one I've encountered this broken.

**Comment by austinbutler** [comment]
Two things stand out for me with Ghostty:

1. It's a [proper GTK application](https://mitchellh.com/writing/ghostty-gtk-rewrite) (and they use the native toolkit for other platforms as well)
   - Here's Wezterm for me, doesn't have any UI to it by default and I can't move it with the mouse
   - <img width="656" height="424" alt="image" src="https://github.com/user-attachments/assets/a18a886b-4289-4757-afa2-2d659f31c43e" />
3. Sometimes they have more than one release in a year :wink: 

<img width="300" alt="image" src="https://github.com/user-attachments/assets/cd4c866f-6860-499d-b430-72e58f297358" />

<img width="300" alt="image" src="https://github.com/user-attachments/assets/a7d057d2-4036-4a9e-98ce-43a052961217" />

Say what you will about Zig or hype or whatever, but when it comes down to it, they are putting in the work. This isn't to say Wezterm is bad because it's not. Just that in addition to an insane amount of hype, there are merits to Ghostty.

**Reply by beipym** [reply]
im using wezterm + nvchad

do you think switching to ghostly is worth it?

( im feeling dumb for asking this )

**Reply by habahabahaba** [reply]
> im using wezterm + nvchad
> 
> do you think switching to ghostly is worth it?
> 
> ( im feeling dumb for asking this )

I can only say, that Wezterm has built-in multiplexer and Ghostty does not. If you're using it, and you don't want to install tmux or Zelij - better stick with Wezterm.

**Reply by lostmygithubaccount** [reply]
> do you think switching to ghostly is worth it?
> 
> ( im feeling dumb for asking this )

I went through this myself recently :) I've always been a MacOS default Terminal.app user, with practically no prompt or other terminal customizations (just a lot of bash aliases). always just used `tmux` for multiplexing, but tried out a new terminal emulator that worked great for a bit and got me hooked on cleaner pane hotkeys that are hard to duplicate in tmux

so I initially choose WezTerm primarily because it's written in Rust, and if I do ever want to contribute I'd prefer that. it was super easy to setup and has worked great for me. but Ghostty also looks promising so I installed it and had Claude Code convert my WezTerm config (which it also wrote) to Ghostyy -- super painless and both emulators seem perfectly fine to me

in my research the big thing people always point out as a key difference is Ghostty uses native GUI elements (e.g. the tabs), whereas WezTerm does not. I actually kinda prefer the WezTerm tabs?

so anyway, both are totally fine options. the lack of releases of WezTerm don't concern me, I'm not sure what additional feature I'd ever need. I do suspect Ghostty will continue picking up more steam and the libghostty work looks very promising; I'm primarily sticking with WezTerm for now myself (but have both installed and configured)

**Reply by sander76** [reply]
One plus for wezterm is it works on windows (which I have to run on because they're twisting my arm)
